src/main/java/gov/nasa/pds/harvest/util/xml/XmlIs.java

```diff
@@ -0,0 +1,69 @@
+package gov.nasa.pds.harvest.util.xml;
+
+import java.io.FileReader;
+import javax.xml.transform.Source;
+import javax.xml.transform.sax.SAXSource;
+import org.xml.sax.InputSource;
+import net.sf.saxon.Configuration;
+import net.sf.saxon.lib.ParseOptions;
+import net.sf.saxon.om.NodeInfo;
+import net.sf.saxon.om.TreeInfo;
+import net.sf.saxon.xpath.XPathEvaluator;
+
+public class XmlIs {
+  private static String ANY_TYPE = "logical_identifier";
+  private static boolean interogate(String filename, String type) {
+    if (filename.endsWith (".xml") || filename.endsWith (".lblx")) {
+      try {
+        XPathEvaluator xpath = new XPathEvaluator();
+        Configuration configuration = xpath.getConfiguration();
+        configuration.setLineNumbering(true);
+        configuration.setXIncludeAware(false);
+        ParseOptions options = new ParseOptions();
+        options.withLineNumbering(true);
+        options.withXIncludeAware(false);
+        Source source = new SAXSource(new InputSource(new FileReader(filename)));
+        TreeInfo docInfo = configuration.buildDocumentTree(source , options);
+        NodeInfo ia=null,lid=null,pcls=null;
+        for (NodeInfo top : docInfo.getRootNode().children()) {
```

Comment:
Using SAX make the code to read the product class a bit complicated, but that is the right approach since we noticed some label can be very big.
Apparently an alternative could be to use StAX , but that might not be worth the work to use that.

Reply:
Should be fast enough as this the approach validate takes. Just reading a couple of nodes right at the top should not be too bad. Even a million nodes loops pretty quick since not descending them all.